app/src/main/java/openfoodfacts/github/scrachx/openfood/views/product/ProductActivity.java

```diff
@@ -54,475 +62,425 @@
 import openfoodfacts.github.scrachx.openfood.views.product.summary.SummaryProductFragment;
+
 import org.json.JSONException;
 import org.json.JSONObject;
 
+import javax.inject.Inject;
+
 import static android.content.pm.PackageManager.PERMISSION_GRANTED;
 import static openfoodfacts.github.scrachx.openfood.utils.Utils.MY_PERMISSIONS_REQUEST_CAMERA;
 
-public class ProductActivity extends BaseActivity implements OnRefreshListener
-{
-
-	private static final int LOGIN_ACTIVITY_REQUEST_CODE = 1;
-	@BindView( R.id.pager )
-	ViewPager viewPager;
-	@BindView( R.id.toolbar )
-	Toolbar toolbar;
-	@BindView( R.id.tabs )
-	TabLayout tabLayout;
-	@BindView( R.id.buttonScan )
-	FloatingActionButton mButtonScan;
-	@BindView( R.id.bottom_navigation )
-	BottomNavigationView bottomNavigationView;
-
-	RecyclerView productBrowsingRecyclerView;
-	ProductFragmentPagerAdapter adapterResult;
-	ProductsRecyclerViewAdapter productsRecyclerViewAdapter;
-
-	private OpenFoodAPIClient api;
-	private ShareActionProvider mShareActionProvider;
-	private State mState;
-	private SensorManager mSensorManager;
-	private Sensor mAccelerometer;
-	private ShakeDetector mShakeDetector;
-	// boolean to determine if scan on shake feature should be enabled
-	private boolean scanOnShake;
-
-	@RequiresApi( api = Build.VERSION_CODES.M )
-	@Override
-	protected void onCreate( Bundle savedInstanceState )
-	{
-		super.onCreate( savedInstanceState );
-		if( getResources().getBoolean( R.bool.portrait_only ) )
-		{
-			setRequestedOrientation( ActivityInfo.SCREEN_ORIENTATION_PORTRAIT );
-		}
-		setContentView( R.layout.activity_product );
-		setTitle( getString( R.string.app_name_long ) );
-
-		setSupportActionBar( toolbar );
-		getSupportActionBar().setDisplayHomeAsUpEnabled( true );
-
-		setupViewPager( viewPager );
-
-		tabLayout.setupWithViewPager( viewPager );
-
-		api = new OpenFoodAPIClient( this );
-
-		mState = (State) getIntent().getExtras().getSerializable( "state" );
-		if( !Utils.isHardwareCameraInstalled( this ) )
-		{
-			mButtonScan.setVisibility( View.GONE );
-		}
-
-		// Get the user preference for scan on shake feature and open ContinuousScanActivity if the user has enabled the feature
-		mSensorManager = (SensorManager) getSystemService( Context.SENSOR_SERVICE );
-		mAccelerometer = mSensorManager.getDefaultSensor( Sensor.TYPE_ACCELEROMETER );
-		mShakeDetector = new ShakeDetector();
-
-		SharedPreferences shakePreference = PreferenceManager.getDefaultSharedPreferences( this );
-		scanOnShake = shakePreference.getBoolean( "shakeScanMode", false );
-
-		mShakeDetector.setOnShakeListener( new ShakeDetector.OnShakeDetected()
-		{
-			@Override
-			public void onShake( int count )
-			{
-
-				if( scanOnShake )
-				{
-					Utils.scan( ProductActivity.this );
-				}
-			}
-		} );
-
-		bottomNavigationView.setOnNavigationItemSelectedListener( item -> {
-
-			switch( item.getItemId() )
-			{
+public class ProductActivity extends BaseActivity implements OnRefreshListener {
```

Comment:
@prashantkh19  Why do you delete the entire thing again and paste it again ? 

This PR has 451 deletions and 417 additions.  Seeing this number it seems that the number of genuine changes that you have made will be less than 50 lines.  

*If the diff in the PR contained only the changes you made, it would be far easier to review. *
 *At this state it is impossible to determine what changes you have made and what was present earlier. *

Reply:
> @prashantkh19 Why do you delete the entire thing again and paste it again ?
> 
> This PR has 451 deletions and 417 additions. Seeing this number it seems that the number of genuine changes that you have made will be less than 50 lines.
> 
> If the diff in the PR contained only the changes you made, it would be far easier to review.
> At this state it is impossible to determine what changes you have made and what was present earlier.

@PrajwalM2212 I am sorry, I am new to open source, didn't think this would create problems. I will take care of this next time.

Reply:
@prashantkh19  The PR should ideally contain only the genuine changes that you have made. This way the diff will show only the changes that you have made to the existing codebase. That way reviewers can validate if your changes are good or not

Reply:
He just reformatted the entire file. It's alright! From next time onwards you can create a separate PR for reformatting if there are a lot of changes regarding the formatting.